translator/conventions/opentelemetry.go

```diff
@@ -0,0 +1,139 @@
+	AttributeNetPeerName  = "net.peer.name" // pull request #349
+	AttributeNetHostIP    = "net.host.ip"   // pull request #349
+	AttributeNetHostPort  = "net.host.port" // pull request #349
+	AttributeNetHostName  = "net.host.name" // pull request #349
+	AttributeIAMUser      = "iam.user"      // proposed in issue #362
+	AttributeIAMRole      = "iam.role"      // proposed in issue #362
+)
+
+// OpenTelemetry Semantic Convention values for component attribute values.
+// Possibly being removed due to issue #336
+const (
+	ComponentTypeHTTP    = "http"
+	ComponentTypeGRPC    = "grpc"
+	ComponentTypeDB      = "db"        // proposed in issue #362
+	ComponentTypeMsg     = "messaging" // proposed in issue #362
+	ComponentTypeCluster = "cluster"   // proposed in issue #362
+)
+
+// OpenTelemetry Semantic Convention attribute names for HTTP related attributes
+// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-http.md
+const (
+	AttributeHTTPMethod     = "http.method"
+	AttributeHTTPURL        = "http.url"
+	AttributeHTTPTarget     = "http.target"
+	AttributeHTTPHost       = "http.host"
+	AttributeHTTPScheme     = "http.scheme"
+	AttributeHTTPStatusCode = "http.status_code"
+	AttributeHTTPStatusText = "http.status_text"
+	AttributeHTTPFlavor     = "http.flavor"
+	AttributeHTTPServerName = "http.server_name"
+	AttributeHTTPPort       = "http.port"
+	AttributeHTTPRoute      = "http.route"
+	AttributeHTTPClientIP   = "http.client_ip"
+	AttributeHTTPUserAgent  = "http.user_agent" // proposed in issue #362, used by OpenCensus and OpenTelemetry
+)
+
+// OpenTelemetry Semantic Convention attribute names for database related attributes
+// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-database.md
+const (
+	AttributeDBType      = "db.type"
+	AttributeDBInstance  = "db.instance"
+	AttributeDBStatement = "db.statement"
+	AttributeDBUser      = "db.user"
+	AttributeDBURL       = "db.url" // pull request #349
+)
+
+// OpenTelemetry Semantic Convention attribute names for gRPC related attributes
+// See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-rpc.md
+const (
+	EventTypeMessage                 = "message"
+	AttributeMessageType             = "message.type"
+	MessageTypeReceived              = "RECEIVED"
+	MessageTypeSent                  = "SENT"
+	AttributeMessageID               = "message.id"
+	AttributeMessageCompressedSize   = "message.compressed_size"
+	AttributeMessageUncompressedSize = "message.uncompressed_size"
+	AttributeMessageContent          = "message.content" // proposed in issue #362
+)
+
+// Proposed OpenTelemetry Semantic Convention attribute names for error/fault/exception related event attributes
+// Derived from OpenTracing: https://github.com/opentracing/specification/blob/master/semantic_conventions.md
+const (
+	EventTypeError        = "error"
```

Comment:
I think we should not mix OpenTelemetty and OpenTracing convention in one package. There may be cases when OpenTelemetry's convention for the same attribute has a different string value (unlikely, but not impossible). Placing them in one package and not have any naming that allows to differentiate between constants defining different sets of conventions can become a source of confusion.

I think it is better to have separate packages for non-OpenTelemetry conventions. Perhaps a good approach would be to use a package name matching the project where the conventions come from, e.g. `translator/conventions/opentracing/attributes.go` and if we do this for symmetry would be best to do the same for OpenTelemetry conventions `translator/conventions/opentelemetry/attributes.go`.

What do you think?

Reply:
I don't think OpenTelemetry will be able to declare the v1 API final until these are defined. I have a pull request open in otep for these attributes. I think we should leave them here and then edit or remove if the final specification does not have them.